### PR TITLE
Add a base metadataExtractor

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/ExtractMetadataOptions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Metadata.ManagedReference
+{
+    using Microsoft.CodeAnalysis;
+    using System.Collections.Generic;
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/ExtractMetadataOptions.cs]: order

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/MetadataExtractor.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Metadata.ManagedReference
+{
+    using Microsoft.CodeAnalysis;
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.DataContracts.ManagedReference;
+    using System;
+    using System.Diagnostics;
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/MetadataExtractor.cs]: order

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/MetadataExtractor.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    internal class MetadataExtractor
+    {
+        private Compilation _compilation;
+        private IAssemblySymbol _assembly;
+        private bool preserveRawComments;
+        private string filterConfigFile;
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/MetadataExtractor.cs]: append new line.

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/ExtractMetadataOptions.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    internal class ExtractMetadataOptions
+    {
+        public bool ShouldSkipMarkup { get; set; }
+        public bool PreserveRawinlineComments { get; set; }
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/ExtractMetadataOptions.cs]: PreserveRaw"I"nlineComments?

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/MetadataExtractor.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.DataContracts.ManagedReference;
+    using System;
+    using System.Diagnostics;
+
+    internal class MetadataExtractor
+    {
+        private Compilation _compilation;
+        private IAssemblySymbol _assembly;
+        private bool preserveRawComments;
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/MetadataExtractor.cs]: naming, all fields start with '_' or not

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**